### PR TITLE
scan: implement connect modal ui (left part)

### LINKED ISSUE #1206: Implement connect modal UI (Left part)
Follow the design
- Implement Left part UI
- Toggle state between Ledger and Mnemonic

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1206

### CODE REVIEW

File: scan/src/components/UserAccount.re [hunk 1]
```diff
@@ -187,10 +311,10 @@ let make = () => {
        | None =>
          <Col>
            <ConnectBtn connect />
-           // TODO: remove later 
+           // TODO: remove later
            <button
              onClick={_ => {
-               dispatchModal(OpenModal(Connect("Test")));
+               dispatchModal(OpenModal(Connect(<ModalLogin />)));
```
Comment (evilpeach) [inline scan/src/components/UserAccount.re]: Let's move all code about `ModalLogin` to `ConnectModal`

File: scan/src/components/modal/ConnectModal.re [hunk 2]
```diff
@@ -12,5 +12,5 @@ module Styles = {
 
 [@react.component]
 let make = (~value) => {
-  <div className=Styles.container> <Text value /> </div>;
+  <div className=Styles.container> value </div>;
```
Comment (evilpeach) [inline scan/src/components/modal/ConnectModal.re]: Implement everything in here.

File: scan/src/contexts/ModalContext.re [hunk 3]
```diff
@@ -1,5 +1,5 @@
 type t =
-  | Connect(string);
+  | Connect(React.element);
```
Comment (evilpeach) [inline scan/src/contexts/ModalContext.re]: Don't pass anything. Just leave it ` | Connect;`

File: scan/src/components/UserAccount.re [hunk 4]
```diff
@@ -159,6 +210,79 @@ module Balance = {
...
+  [@react.component]
+  let make = (~name, ~active) => {
+    <div className={Styles.buttonContainer(active)}>
+      <div className={Styles.header(active)}>
+        <Text value={name |> toLoginMethodString} weight=Text.Medium size=Text.Md />
+        {switch (name) {
+         | Ledger =>
+           active
+             ? <img src=Images.ledgerIconActive className=Styles.ledgerIcon />
+             : <img src=Images.ledgerIconInactive className=Styles.ledgerIcon />
```
Comment (evilpeach) [inline scan/src/components/UserAccount.re]: Let's use a single image and control opacity when active or not.

**Review by evilpeach — CHANGES_REQUESTED** [review]
Add more padding-top of "Select your ..". Please recheck the design.

File: scan/src/components/modal/ConnectModal.re [hunk 5]
```diff
@@ -1,16 +1,138 @@
...
+  switch (method) {
+  | Mnemonic => "Mnemonic Phrase"
+  | Ledger => "Ledger"
+  };
+};
+
+module LoginMethod = {
+  [@react.component]
+  let make = (~name, ~active) => {
+    <div className={Styles.buttonContainer(active)}>
```
Comment (evilpeach) [inline scan/src/components/modal/ConnectModal.re]: add onClick as props

File: scan/src/components/modal/ConnectModal.re [hunk 6]
```diff
@@ -1,16 +1,138 @@
...
+      </div>
+      <Row>
+        <div className=Styles.itemCol>
+          {[|Mnemonic, Ledger|]
+           ->Belt_Array.map(method =>
+               <div>
+                 <VSpacing size=Spacing.md />
+                 <VSpacing size=Spacing.xs />
+                 <div onClick={_ => setLoginMethod(_ => method)}>
+                   <LoginMethod name=method active={loginMethod == method} />
```
Comment (evilpeach) [inline scan/src/components/modal/ConnectModal.re]: pass `onClick` to `LoginMethod`

File: scan/src/components/modal/ConnectModal.re [hunk 7]
```diff
@@ -1,16 +1,138 @@
...
+                 </div>
+               </div>
+             )
+           ->React.array}
+        </div>
+        <HSpacing size=Spacing.lg />
+        <HSpacing size=Spacing.sm />
+        <div className=Styles.itemCol> <div className=Styles.seperatedLongLine /> </div>
+        <HSpacing size=Spacing.lg />
+        <HSpacing size=Spacing.sm />
```
Comment (evilpeach) [inline scan/src/components/modal/ConnectModal.re:139]: Let's use `marginLeft` and `marginRight`  or `margin2` on `Styles. seperatedLongLine`.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — DISMISSED** [review]

**Review by prin-r — CHANGES_REQUESTED** [review]
Screen Shot 2563-04-21 at 13 17 54

**Review by evilpeach — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Pull request merged**